client/web/src/insights/pages/dashboards/dashboard-page/components/delete-dashboard-modal/DeleteDashboardModal.tsx

```diff
@@ -0,0 +1,82 @@
+import Dialog from '@reach/dialog'
+import { VisuallyHidden } from '@reach/visually-hidden'
+import classnames from 'classnames'
+import CloseIcon from 'mdi-react/CloseIcon'
+import React, { useRef } from 'react'
+import { useHistory } from 'react-router-dom'
+
+import { isErrorLike } from '@sourcegraph/codeintellify/lib/errors'
+import { PlatformContextProps } from '@sourcegraph/shared/src/platform/context'
+
+import { ErrorAlert } from '../../../../../../components/alerts'
+import { LoaderButton } from '../../../../../../components/LoaderButton'
+import { SettingsBasedInsightDashboard } from '../../../../../core/types'
+
+import styles from './DeleteDashobardModal.module.scss'
+import { useDeleteDashboardHandler } from './hooks/use-delete-dashboard-handler'
+
+export interface DeleteDashboardModalProps extends PlatformContextProps<'updateSettings'> {
+    dashboard: SettingsBasedInsightDashboard
+    onClose: () => void
+}
+
+/**
+ * Displays dashboard delete modal window.
+ */
+export const DeleteDashboardModal: React.FunctionComponent<DeleteDashboardModalProps> = props => {
+    const { dashboard, platformContext, onClose } = props
+    const history = useHistory()
+
+    // Remember dashboard in first render of this component.
+    // We have to have a kind of snapshot of dashboard cause in updating handler
+    // we remove this dashboard from the settings cascade and dashboard prop will be equal
+    // to undefined at some point but we have to use dashboard info to redirect user
+    const dashboardReference = useRef(dashboard)
+
+    const handleDeleteSuccess = (): void => {
+        history.push(`/insights/dashboards/${dashboardReference.current.owner.id}`)
+        onClose()
+    }
+
+    const { loadingOrError, handler } = useDeleteDashboardHandler({
+        platformContext,
+        dashboard,
+        onSuccess: handleDeleteSuccess,
+    })
+
+    const isDeleting = !isErrorLike(loadingOrError) && loadingOrError
+
+    return (
+        <Dialog className={styles.modal} onDismiss={onClose} aria-label="Delete code insight dashboard modal">
+            <button type="button" className={classnames('btn btn-icon', styles.closeButton)} onClick={onClose}>
+                <VisuallyHidden>Close</VisuallyHidden>
+                <CloseIcon />
+            </button>
+
+            <h2 className="text-danger">Delete "{dashboard.title}"</h2>
+
+            <span className="d-block mb-4">
+                This can't be undone. You will still be able to access insights from this dashboard in "All insights”.
```

Comment:
Consistent quotes: 

```suggestion
                This can't be undone. You will still be able to access insights from this dashboard in ”All insights”.
```

Also, it might be good to use the same type of quotes in the `h2` label above.

Reply:
Thanks!